### ISSUE TITLE
No apps can perform this action

### ISSUE THREAD

**Issue by Nyankoo** [yasirkula/UnityNativeGallery#243]
**Description of the bug**

When calling the following code, one of our users gets the OS popup "No apps can perform this action":
`NativeGallery.GetImageFromGallery(onAvatarSelected, "Pick your avatar!", "image/jpeg");`

**Reproduction steps**

We can't reproduce this on our end, only happens to the user that reported it from our app.

**Platform specs**

- Unity version: 2020.3.29f1
- Platform: Android 9
- Device: U307AS (Wiko Life 2)
- How did you download the plugin: Package Manager (version 1.6.7)

**Comment by tokkoro** [comment]
I have this same behaviour on my own Android phone. I try to call `NativeGallery.GetVideoFromGallery(callback);` after I have checked permissions. First device screenlocks and then when I unlock it and press the button again that calls the `GetVideoFromGallery` I get "No apps can perform this action" (in Finnish).

ADB logs doesn't say anything. I'm not sure where to continue.

![image](https://user-images.githubusercontent.com/20582490/187061925-68874bfa-4a73-4f9c-b1a0-3c9d73b8598b.png)

**Comment by yasirkula** [comment]
You can try calling the following in Awake/Start:

```csharp
using( AndroidJavaClass ajc = new AndroidJavaClass( "com.yasirkula.unity.NativeGalleryMediaPickerFragment" ) )
	ajc.SetStatic<bool>( "preferGetContent", true );
```

**Comment by tokkoro** [comment]
Thank you @yasirkula that fixed it to me. Was that mentioned somewhere else that I missed (should I have searched harder)? Is that something that can/will be fixed inside the library or should I make a PR to add it into Readme? I'm not confortable to make a PR to the library code it self as I'm not too familiar what else changing the `preferGetContent` would do.

**Comment by yasirkula** [comment]
@tokkoro NativeGallery uses ACTION_PICK intent by default because it works better with gallery IMO. I'd expect all Android devices to support ACTION_PICK intent but perhaps some rooted phones that've removed the built-in gallery app or phones from some device vendors where they've forgotten to put ACTION_PICK support to their stock gallery apps are exceptions. ACTION_GET_CONTENT is another method of picking files from gallery but its UI isn't as easy to grasp as ACTION_PICK for a user IMO. I might just fallback to ACTION_GET_CONTENT if ACTION_PICK can't find any apps to perform that action.

**Comment by tokkoro** [comment]
Sounds like the fragmentation of Android ecosystem strikes again :(

For the record, the phone is non-rootted One Plus 9 (LE2113), Android 12, with semi latest Oxygen OS (LE2113_11_C.48)  and with just the gallery app that is the one preinstalled.

**Comment by yasirkula** [comment]
Thank you for providing these information! I'll post an updated native library file here for testing purposes when I apply the change I've mentioned.

**Comment by yasirkula** [comment]
@tokkoro Does replacing NativeGallery.aar with the one inside [this zip archive](https://github.com/yasirkula/UnityNativeGallery/files/9620041/NativeGallery.zip) resolve the issue without having to set "preferGetContent" to true?

**Comment by tokkoro** [comment]
I uninstalled current version from Unity Package manager and then downloaded and imported the unitypackage. After that I downloaded and replaced the [custom .aar file](https://github.com/yasirkula/UnityNativeGallery/files/9620041/NativeGallery.zip) you send in your previous message.

I also commented out the previous fix `ajc.SetStatic<bool>("preferGetContent", true);`

After updating the version number to make sure I have the correct version installed on my phone I finally build and run the new version of the game and it **works**. Looks way better that the `preferGetContent` work around.

I have also moved to Unity 2021 as the 2019 version had nasty crashing bug with playing video with Unity's own `VideoPlayer` component.

So at least for my setup this update for NativeGallery fixes the issue!
My setup: 
Unity 2021.3.10f1,
Android minimum 5.1 (api 22)
Android target: Highest installed (api 33)

☕  Thank you @yasirkula I purchased a large coffee for you, I suggest that anyone else benefiting from these well made and _mainteined_ unity packages does so also. Info is in the [bio ](https://github.com/yasirkula) as always.
